### DISCUSSION TITLE
Grafana 6.4.2 Query as variable in link url

### DISCUSSION THREAD

**Discussion by fredbcode** [grafana/grafana#38348] [Dashboards, Panels & Visualizations]
Hello all,

I'm using grafana with elasticsearch source and all works very well
I added an url link when a result appears, ok

But I like to add an url to my kibana/elastic with the query as parameters, and it seems I can't now

I guess this can be also very useful for other solution, ticketing, etc

https://myelastic.local/blalabbla **${_value,query}**

**Comment by papagian** [comment]
Could you be a bit more specific?
How do you try to add the link and what are you trying to achieve?
Also, which version of grafana and elasticsearch do you use?
thank you

**Comment by fredbcode** [comment]
Of course

I have a dashboard with multi queries (Elasticsearch as source), each queries is a different request to my Kibana/elasticsearch cluster

A) message:"ERROR"
B) hostname=test
C) message:"Down"

etc

In visualization I added a data links to my kibana, so the user can click to switch to the Kibana/elasticsearch website and see what exactly happened 

Unfortunately I found no way to pass the query as variable 

http://kibana.local/xxxx?QUERYxxx (where query is the value of A,B,C ...)


Beyond my usage, I think this can be very useful to pass the query as request to another site

For example also for ticketing purpose, eg with Prometheus as source  

http://ticketing.local/create-ticket.php?probe_success{instance="toto.com"}?... 

When it's necessary the operator click on his panel and generates a ticket with predefined value (time, node, etc, and also the query who generated an alert)

Thank you

**Comment by papagian** [comment]
Closing this since it is something that needs to be supported by Kibana instead of Grafana.

**Comment by fredbcode** [comment]
Why ? 

Kibana just receive a link from browser generated by Grafana
If the Query in url is missing what Kibana can do ?

**Comment by papagian** [comment]
@fredbcode
Sorry, I misinterpreted your reply; I thought that the query is passed but it is not handled by Kibana.

**Comment by fredbcode** [comment]
No problem, English is not my native language ...

Pass the query to an url (made by grafana) could be useful for many tools, I guess

**Comment by e-d-i-t** [comment]
Exactly, just running the first 2 minutes on Grafana in my life and need it right away. For instance... I got pfsense logs in dashboard and the current most blocked IP address want to pass through a GET-statement link which calls a whois website along with that IP address...

**Comment by e-d-i-t** [comment]
And a few minutes later ;-)
Found it. create a datalink and just use the right formatted field. Eventually it works.
Now I see an IP address in a panel, click it, go to another webpage which performs the whois for me. NICE!

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #83007. Feel free to continue the discussion around this there. Thank you!
